Index: includes/common.inc
===================================================================
--- includes/common.inc	(revision 22)
+++ includes/common.inc	(working copy)
@@ -1307,6 +1307,7 @@
  *   A translated date string in the requested format.
  */
 function format_date($timestamp, $type = 'medium', $format = '', $timezone = NULL, $langcode = NULL) {
+  if(function_exists('dst_format_date') ) return dst_format_date($timestamp, $type, $format, $timezone, $langcode);
   if (!isset($timezone)) {
     global $user;
     if (variable_get('configurable_timezones', 1) && $user->uid && strlen($user->timezone)) {
