Index: .htaccess
===================================================================
--- .htaccess
+++ .htaccess
@@ -123,6 +123,41 @@
   RewriteCond %{REQUEST_URI} !^404.%1$
   RewriteRule ^(.*)$ 404.%1 [L]
 
+  # HTTPS redirect policy. [#3470]
+  RewriteCond %{HTTPS} on
+  RewriteCond %{REQUEST_URI}      !^/(user|dolacz-do-nas|przypomnij-haslo)$
+  RewriteCond %{QUERY_STRING}     !^q=(user|dolacz-do-nas|przypomnij-haslo)
+  RewriteCond %{REQUEST_URI}      !^/user(/login$|/register$|/password$|/reset/.*|/.*/edit|/.*/roles)
+  RewriteCond %{QUERY_STRING}     !^q=user(/login$|/register$|/password$|/reset/.*|/.*/edit|/.*/roles)
+  RewriteCond %{REQUEST_URI}      !^/autor/.*/(edit|roles)
+  RewriteCond %{QUERY_STRING}     !^q=autor/.*/(edit|roles)
+  RewriteCond %{REQUEST_URI}      !^/admin(|/.*)
+  RewriteCond %{QUERY_STRING}     !^q=admin(|/.*)
+  RewriteCond %{REQUEST_URI}      !^/js(|/.*)
+  RewriteCond %{QUERY_STRING}     !^q=js(|/.*)
+  RewriteCond %{REQUEST_URI}      !^(|/.*)ajax(|/.*)
+  RewriteCond %{QUERY_STRING}     !^q=(|/.*)ajax(|/.*)
+  RewriteCond %{REQUEST_URI}      !^(|/.*)ahah(|/.*)
+  RewriteCond %{QUERY_STRING}     !^q=(|/.*)ahah(|/.*)
+  RewriteCond %{REQUEST_URI}      !^(|/.*)autocomplete(|/.*)
+  RewriteCond %{QUERY_STRING}     !^q=(|/.*)autocomplete(|/.*)
+  RewriteCond %{REQUEST_URI}      !^/upload_element_js/.*
+  RewriteCond %{QUERY_STRING}     !^q=upload_element_js|/.*
+  RewriteRule ^(.*) http://%{HTTP_HOST}%{REQUEST_URI} [R=301,L]
+
+  RewriteCond %{HTTP_HOST} !^www.mm360.pl$
+  RewriteCond %{HTTPS} off
+  RewriteCond %{REQUEST_URI} ^/(user|dolacz-do-nas|przypomnij-haslo)$ [OR]
+  RewriteCond %{REQUEST_URI} ^/user(/login$|/register$|/password$|/reset/.*|/.*/edit|/.*/roles) [OR]
+  RewriteCond %{REQUEST_URI} ^/autor/.*/(edit|roles) [OR]
+  RewriteCond %{REQUEST_URI} ^/admin(|/.*)
+  RewriteCond %{REQUEST_URI} !^/js(|/.*)
+  RewriteCond %{REQUEST_URI} !^(|/.*)ajax(|/.*)
+  RewriteCond %{REQUEST_URI} !^(|/.*)ahah(|/.*)
+  RewriteCond %{REQUEST_URI} !^(|/.*)autocomplete(|/.*)
+  RewriteCond %{REQUEST_URI} !^/upload_element_js/.*
+  RewriteRule ^(.*) https://%{HTTP_HOST}%{REQUEST_URI} [R=301,L]
+
   # Rewrite URLs of the form 'x' to the form 'index.php?q=x'.
   RewriteCond %{REQUEST_FILENAME} !-f
   RewriteCond %{REQUEST_FILENAME} !-d
