Index: includes/bootstrap.inc
===================================================================
--- includes/bootstrap.inc	(revision 6988)
+++ includes/bootstrap.inc	(working copy)
@@ -783,7 +783,7 @@
     }
 
     header("Expires: Sun, 19 Nov 1978 05:00:00 GMT");
-    header("Cache-Control: must-revalidate");
+    header("Cache-Control: no-store, no-cache, must-revalidate");
   }
 
   // Send appropriate response:
