Index: app/modules/node/node.module
===================================================================
--- app/modules/node/node.module	(revision 7202)
+++ app/modules/node/node.module	(working copy)
@@ -1887,7 +1887,14 @@
     );
 
     // Taxonomy box:
-    if ($taxonomy = module_invoke('taxonomy', 'form_all', 1)) {
+    // #4972. This module executes taxonomy_form_all with free tags argument = 0.
+    // This options is real memory hog, if we have (and we have) huge tags
+    // categories. Becouse php is lazy at condition verification, this
+    // variable_get call will prevent problems with out of memory
+    if (
+      !variable_get('search_override_no_category', TRUE) &&
+      $taxonomy = module_invoke('taxonomy', 'form_all', FALSE)
+    ) {
       $form['advanced']['category'] = array(
         '#type' => 'select',
         '#title' => t('Only in the category(s)'),
