Index: app/sites/all/modules/contrib/views/includes/view.inc
===================================================================
--- includes/view.inc	(revision 2682)
+++ includes/view.inc	(working copy)
@@ -692,8 +692,10 @@
       $function($this);
     }
 
-    $query = db_rewrite_sql($this->build_info['query'], $this->base_table, $this->base_field, array('view' => &$this));
-    $count_query = db_rewrite_sql($this->build_info['count_query'], $this->base_table, $this->base_field, array('view' => &$this));
+    if (!empty($this->build_info['query'])) {
+      $query = db_rewrite_sql($this->build_info['query'], $this->base_table, $this->base_field, array('view' => &$this));
+      $count_query = db_rewrite_sql($this->build_info['count_query'], $this->base_table, $this->base_field, array('view' => &$this));
+    }
 
     $args = $this->build_info['query_args'];
 
