Index: sites/all/modules/contrib/views/includes/query.inc
===================================================================
--- sites/all/modules/contrib/views/includes/query.inc	(wersja 2371)
+++ sites/all/modules/contrib/views/includes/query.inc	(wersja 2463)
@@ -980,7 +980,14 @@
     }
 
     if ($has_aggregate || $this->groupby) {
-      $groupby = "GROUP BY " . implode(', ', array_unique(array_merge($this->groupby, $non_aggregates))) . "\n";
+
+      //$groupby = "GROUP BY " . implode(', ', array_unique(array_merge($this->groupby, $non_aggregates))) . "\n";
+      if (!empty($this->groupby)) {
+        $groupby = "GROUP BY " . implode(', ', array_unique($this->groupby)) . "\n";
+      }
+      else {
+        $groupby = "GROUP BY " . implode(', ', array_unique(array_merge($this->groupby, $non_aggregates))) . "\n";
+      }
       if ($this->having) {
         $having = $this->condition_sql('having');
       }
