Index: jscripts/tiny_mce/themes/advanced/langs/pl_dlg.js
===================================================================
--- jscripts/tiny_mce/themes/advanced/langs/pl_dlg.js	(wersja 8322)
+++ jscripts/tiny_mce/themes/advanced/langs/pl_dlg.js	(kopia robocza)
@@ -38,6 +38,7 @@
 image_align_texttop:"G\u00F3rny tekst",
 image_align_textbottom:"Dolny tekst",
 image_align_left:"Lewy",
+link_rel:"Atrybut rel",
 image_align_right:"Prawy",
 link_title:"Wstaw/edytuj link",
 link_url:"Link URL",
Index: jscripts/tiny_mce/themes/advanced/link.htm
===================================================================
--- jscripts/tiny_mce/themes/advanced/link.htm	(wersja 8322)
+++ jscripts/tiny_mce/themes/advanced/link.htm	(kopia robocza)
@@ -37,6 +37,10 @@
 			<td><label id="targetlistlabel" for="targetlist">{#advanced_dlg.link_target}</label></td>
 			<td><select id="target_list" name="target_list"></select></td>
 		</tr>
+		<tr>
+			<td><label id="rellistlabel" for="rellist">{#advanced_dlg.link_rel}</label></td>
+			<td><select id="rel_list" name="rel_list"></select></td>
+		</tr>
           <tr>
             <td class="nowrap"><label for="linktitle">{#advanced_dlg.link_titlefield}</label></td>
             <td><input id="linktitle" name="linktitle" type="text" value="" style="width: 200px" /></td>
Index: jscripts/tiny_mce/themes/advanced/js/link.js
===================================================================
--- jscripts/tiny_mce/themes/advanced/js/link.js	(wersja 8322)
+++ jscripts/tiny_mce/themes/advanced/js/link.js	(kopia robocza)
@@ -19,6 +19,7 @@
 		this.fillClassList('class_list');
 		this.fillFileList('link_list', 'tinyMCELinkList');
 		this.fillTargetList('target_list');
+		this.fillRelList('rel_list');
 
 		if (e = ed.dom.getParent(ed.selection.getNode(), 'A')) {
 			f.href.value = ed.dom.getAttrib(e, 'href');
@@ -27,6 +28,7 @@
 			selectByValue(f, 'link_list', f.href.value);
 			selectByValue(f, 'target_list', ed.dom.getAttrib(e, 'target'));
 			selectByValue(f, 'class_list', ed.dom.getAttrib(e, 'class'));
+			selectByValue(f, 'rel_list', ed.dom.getAttrib(e, 'rel'));
 		}
 	},
 
@@ -64,7 +66,8 @@
 						href : href,
 						title : f.linktitle.value,
 						target : f.target_list ? getSelectValue(f, "target_list") : null,
-						'class' : f.class_list ? getSelectValue(f, "class_list") : null
+						'class' : f.class_list ? getSelectValue(f, "class_list") : null,
+						rel : f.rel_list ? getSelectValue(f, "rel_list") : null,
 					});
 				}
 			});
@@ -73,7 +76,8 @@
 				href : href,
 				title : f.linktitle.value,
 				target : f.target_list ? getSelectValue(f, "target_list") : null,
-				'class' : f.class_list ? getSelectValue(f, "class_list") : null
+				'class' : f.class_list ? getSelectValue(f, "class_list") : null,
+				rel : f.rel_list ? getSelectValue(f, "rel_list") : null,
 			});
 		}
 
@@ -149,6 +153,19 @@
 				lst.options[lst.options.length] = new Option(v[0], v[1]);
 			});
 		}
+	},
+	fillRelList : function(id) {
+		var dom = tinyMCEPopup.dom, lst = dom.get(id), v;
+
+		lst.options[lst.options.length] = new Option(tinyMCEPopup.getLang('not_set'), '');
+		lst.options[lst.options.length] = new Option('nofollow', 'nofollow');
+
+		if (v = tinyMCEPopup.getParam('theme_advanced_link_targets')) {
+			tinymce.each(v.split(','), function(v) {
+				v = v.split('=');
+				lst.options[lst.options.length] = new Option(v[0], v[1]);
+			});
+		}
 	}
 };
 
