Index: js/gmap.js
===================================================================
--- js/gmap.js	(revision 6198)
+++ js/gmap.js	(working copy)
@@ -329,6 +329,7 @@
           map.setCenter(new GLatLng(obj.vars.latitude, obj.vars.longitude), obj.vars.zoom);
         };
         $(elem).parents('fieldset.collapsible').children('legend').children('a').click(r);
+        $('.vertical-tab-button').children('a').click(r);
         // Would be nice, but doesn't work.
         //$(elem).parents('fieldset.collapsible').children('.fieldset-wrapper').scroll(r);
       }, 0);
