Index: memcache_admin.module
===================================================================
--- memcache_admin.module	(revision 8053)
+++ memcache_admin.module	(working copy)
@@ -225,6 +225,8 @@
  * @return string
  */
 function memcache_admin_stats($bin = 'cache') {
+  module_load_include('inc', 'memcache', 'dmemcache');
+
   $bin = memcache_admin_bin_mapping($bin);
 
   $stats = dmemcache_stats($bin, 'default', TRUE);
@@ -408,6 +410,8 @@
 }
 
 function memcache_admin_stats_types($bin) {
+  module_load_include('inc', 'memcache', 'dmemcache');
+
   if ($mc = dmemcache_object($bin)) {
     if ($mc instanceof Memcache) {
       // TODO: Determine which versions of the PECL memcache extension have
