Index: filefield_widget.inc
===================================================================
--- filefield_widget.inc
+++ filefield_widget.inc
@@ -536,13 +536,13 @@
     }
 
     if (!$valid && $field['required']) {
-      form_set_error($field_name, t('%title field is required.', array('%title' => $field['widget']['label'])));
+      form_set_error($field_name, t('%title field is required.', array('%title' => t($field['widget']['label']))));
     }
     $max_filesize = parse_size($field['widget']['max_filesize_per_node']);
     if ($max_filesize && $total_filesize > $max_filesize) {
       form_set_error($field_name, t('Total filesize for %title, %tsize, exceeds field settings of %msize.',
                                     array(
-                                      '%title' => $field['widget']['label'],
+                                      '%title' => t($field['widget']['label']),
                                       '%tsize' => format_size($total_filesize),
                                       '%msize' => format_size($max_filesize)
                                     )
