--- tinymce.inc  2011-07-12 10:07:43.151717642 +0200
+++ tinymce.inc 2011-07-07 15:31:34.000000000 +0200
@@ -84,7 +84,7 @@
   $script = $editor['library path'] . '/tiny_mce.js';
   $script = fopen($script, 'r');
   // Version is contained in the first 100 chars.
-  $line = fgets($script, 100);
+  $line = fgets($script, 150);
   // 2.x: this.majorVersion="2";this.minorVersion="1.3"
   // 3.x: majorVersion:'3',minorVersion:'2.0.1'
   if (preg_match('@majorVersion[=:]["\'](\d).+?minorVersion[=:]["\']([\d\.]+)@', $line, $version)) {