### Eclipse Workspace Patch 1.0
#P drupal.org.ctools
Index: plugins/content_types/node_context/node_comments.inc
===================================================================
RCS file: /cvs/drupal-contrib/contributions/modules/ctools/plugins/content_types/node_context/node_comments.inc,v
retrieving revision 1.2.2.3
diff -u -r1.2.2.3 node_comments.inc
--- plugins/content_types/node_context/node_comments.inc	14 Apr 2010 20:46:53 -0000	1.2.2.3
+++ plugins/content_types/node_context/node_comments.inc	2 Jun 2010 12:01:36 -0000
@@ -95,7 +95,15 @@
 
   // Multiple comment view
   $query_count = 'SELECT COUNT(*) FROM {comments} c WHERE nid = %d';
-  $query = 'SELECT c.cid AS cid, c.pid, c.nid, c.subject, c.comment, c.format, c.timestamp, c.name, c.mail, c.homepage, u.uid, u.name AS registered_name, u.picture, u.data, c.thread, c.status, parent_user.uid as parent_uid, parent_user.data as parent_data, parent_user.name as parent_name, parent_user.picture as parent_picture FROM {comments} c INNER JOIN {users} u ON c.uid = u.uid LEFT OUTER JOIN {comments} parent ON c.pid = parent.cid LEFT OUTER JOIN {users} parent_user ON parent.uid = parent_user.uid WHERE c.nid = %d';
+
+  if (db_column_exists('users', 'signature_format')) {
+    //We run drupal version > 6.13, users table has signature_format column
+    $query = 'SELECT c.cid AS cid, c.pid, c.nid, c.subject, c.comment, c.format, c.timestamp, c.name, c.mail, c.homepage, u.uid, u.name AS registered_name, u.picture, u.signature, u.signature_format, u.data, c.thread, c.status, parent_user.uid as parent_uid, parent_user.data as parent_data, parent_user.name as parent_name, parent_user.picture as parent_picture FROM {comments} c INNER JOIN {users} u ON c.uid = u.uid LEFT OUTER JOIN {comments} parent ON c.pid = parent.cid LEFT OUTER JOIN {users} parent_user ON parent.uid = parent_user.uid WHERE c.nid = %d';
+  }
+  else {
+    //We run outdated drupal version <= 6.13, users table doesn't have signature_format column
+    $query = 'SELECT c.cid AS cid, c.pid, c.nid, c.subject, c.comment, c.format, c.timestamp, c.name, c.mail, c.homepage, u.uid, u.name AS registered_name, u.picture, u.signature, u.data, c.thread, c.status, parent_user.uid as parent_uid, parent_user.data as parent_data, parent_user.name as parent_name, parent_user.picture as parent_picture FROM {comments} c INNER JOIN {users} u ON c.uid = u.uid LEFT OUTER JOIN {comments} parent ON c.pid = parent.cid LEFT OUTER JOIN {users} parent_user ON parent.uid = parent_user.uid WHERE c.nid = %d';
+  }
 
   $query_args = array($node->nid);
   if (!user_access('administer comments')) {
@@ -122,7 +130,7 @@
   }
   $query = db_rewrite_sql($query, 'c', 'cid');
   $query_count = db_rewrite_sql($query_count, 'c', 'cid');
-  
+
   // Start a form, for use with comment control.
   $result = pager_query($query, $comments_per_page, 0, $query_count, $query_args);
 
